sdk/storage/azblob/blob/client_test.go

```diff
@@ -252,6 +254,73 @@ func (s *BlobUnrecordedTestsSuite) TestCopyBlockBlobFromUrlSourceContentMD5() {
 	_require.NotNil(err)
 }
 
+// This test simulates DownloadFile/Buffer methods,
+// and verifies length and content of file
+func (s *BlobUnrecordedTestsSuite) TestUploadDownloadBlockBlob() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	if err != nil {
+		s.Fail("Unable to fetch service client because " + err.Error())
+	}
+
+	containerName := testcommon.GenerateContainerName(testName)
+	containerClient := testcommon.CreateNewContainer(context.Background(), _require, containerName, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	const MiB = 1024 * 1024
+	testUploadDownload := func(contentSize int) {
+		content := make([]byte, contentSize)
+		_, _ = rand.Read(content)
+		contentMD5 := md5.Sum(content)
+		body := streaming.NopCloser(bytes.NewReader(content))
+
+		srcBlob := containerClient.NewBlockBlobClient("srcblob")
+
+		// Prepare source bbClient for copy.
+		_, err = srcBlob.Upload(context.Background(), body, nil)
+		_require.Nil(err)
+
+		// downlod to a temp file and verify contents
+		tmp, err := os.CreateTemp("", "")
+		_require.Nil(err)
+		defer tmp.Close()
+
+		f := blob.DownloadFileOptions{BlockSize: 2 * MiB}
+		n, err := srcBlob.DownloadFile(context.Background(), tmp, &f)
+		_require.Nil(err)
+		_require.Equal(int64(contentSize), n)
+
+		// Compute md5 of file, and verify it against stored value.
+		_, _ = tmp.Seek(0, io.SeekStart)
+		buff := make([]byte, contentSize)
+		_, err = io.ReadFull(tmp, buff)
+		_require.Nil(err)
+		_require.Equal(contentMD5, md5.Sum(buff))
+
+		// Download to a buffer and verify contents
+		buff = make([]byte, contentSize)
+		b := blob.DownloadBufferOptions{BlockSize: 2 * MiB}
+		n, err = srcBlob.DownloadBuffer(context.Background(), buff, &b)
+		_require.Nil(err)
+		_require.Equal(int64(contentSize), n)
+		_require.Equal(contentMD5, md5.Sum(buff[:]))
```

Comment:
We are validating the downloaded file with two options I see one with DownlloadBuffer and one with DownloadFile. New code of serial write is not executed for download buffer correct ?

Reply:
New code will not be executed for DownloadBuffer. However, this function was also untested. And both of them can be tested similarly.